### DISCUSSION TITLE
Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.

### DISCUSSION THREAD

**Discussion by phillyoon** [vercel/next.js#46189] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.1.0: Sun Oct  9 20:15:52 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T8112
    Binaries:
      Node: 19.6.1
      npm: 9.4.0
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.1.6
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0

### Link to the code that reproduces this issue

https://github.com/DevFlipp/postitMAIN

### To Reproduce

Go to Vercel and try deploying it. You're gonna basically see "Application error: a client-side exception has occurred (see the browser console for more information)." and in console "Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error."

### Describe the Bug

When using localhost:3000 there are no errors, but when actually trying to deploy it it just won't and it displays that console error.

### Expected Behavior

For the website to actually load.

### Which browser are you using? (if relevant)

Brave

### How are you deploying your application? (if relevant)

Vercel

**Comment by balazsorban44** [comment]
Could you specify on which page you see this error? Does this also happen when running `next build && next start` locally and then visiting `http://localhot:3000` or only when deployed to Vercel?

**Reply by phillyoon** [reply]
Both it displays "Application error: a client-side exception has occurred (see the browser console for more information)." on the page and in the developer console this is displayed "Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.".

**Reply by icyJoseph** [reply]
```console
at e.SessionProvider (https://postit-main.vercel.app/_next/static/chunks/299-89a2b98c3d7197ba.js:1:8484)
    at s (https://postit-main.vercel.app/_next/static/chunks/app/layout-657901e572716856.js:1:636)
        at h (https://postit-main.vercel.app/_next/static/chunks/481-315ef772b4b20ad3.js:9:31263)
            at l (https://postit-main.vercel.app/_next/static/chunks/app/layout-657901e572716856.js:1:404)
                at body    at html    at v (https://postit-main.vercel.app/_next/static/chunks/17-f95535aff69e85a7.js:1:8018)
                    at l (https://postit-main.vercel.app/_next/static/chunks/17-f95535aff69e85a7.js:1:13959)
                        at t.ErrorBoundary (https://postit-main.vercel.app/_next/static/chunks/17-f95535aff69e85a7.js:1:13008)
                            at t.default (https://postit-main.vercel.app/_next/static/chunks/17-f95535aff69e85a7.js:1:7597)
                                at k (https://postit-main.vercel.app/_next/static/chunks/17-f95535aff69e85a7.js:1:5759)
                                    at C    at P (https://postit-main.vercel.app/_next/static/chunks/17-f95535aff69e85a7.js:1:6099)
```

**Reply by balazsorban44** [reply]
The repo seems to be private (get a 404), so I cannot look at the source code, @icyJoseph where did you get that Vercel deployment from? :eyes:

**Reply by icyJoseph** [reply]
It was a public repo before, the deployment is gone too https://postit-main.vercel.app/  😕 

I got the log info by placing debuggers where the errors happened and stepped through till I got some meaningful message. 

Anyhow, if OP has taken this down, there might be a very good reason for it.

**Comment by alishaaaa1920** [comment]
Did you solve your problem ? Cause i'm getting it noww

**Reply by VanThanh258** [reply]
The solution you propose—what is it? Could you share it?

**Reply by gustaflugnegard** [reply]
I solved this problem by moving my create server client (to verify auth user, ie a private page) to my page.tsx instead of defining it in the layout.tsx. Worked great in dev but not so much in prod. Dont know why.
